### ISSUE TITLE
bigtable: unknown family error hidden (and retried forever) in ApplyBulk

### ISSUE THREAD

**Issue by Iulian7** [GoogleCloudPlatform/google-cloud-go#611]
When I apply mutations with an unknown column family (that has not been created yet) `ApplyBulk` never returns. Here is a self contained test case: 

```
func TestUnknownFamily(t *testing.T) {
	adminClient, client, c := makeClient(t)
	defer c()

	ctx := context.Background()

	if err := adminClient.CreateTable(ctx, "table"); err != nil {
		t.Fatalf("creating table: %v", err)
	}
	tbl := client.Open("table")

	rowKeys := []string{"k1", "k2"}
	muts := make([]*bigtable.Mutation, len(rowKeys))
	for i, k := range rowKeys {
		muts[i] = bigtable.NewMutation()
		muts[i].Set("family", "col", bigtable.Now(), []byte(k))
	}

	rowErrs, err := tbl.ApplyBulk(ctx, rowKeys, muts)
	if err != nil {
		t.Errorf("apply bulk failed: %v", err)
	}
	if len(rowErrs) == 0 {
		t.Error("no row errors with unknown family")
	}
}
```
The log output is misleading: 

```
2017/04/22 21:21:36 Retryable error: rpc error: code = DeadlineExceeded desc = Synthetic error: partial failure of ApplyBulk, retrying in 47.77941ms
2017/04/22 21:21:36 Retryable error: rpc error: code = DeadlineExceeded desc = Synthetic error: partial failure of ApplyBulk, retrying in 2.153551ms
2017/04/22 21:21:36 Retryable error: rpc error: code = DeadlineExceeded desc = Synthetic error: partial failure of ApplyBulk, retrying in 114.145821ms
2017/04/22 21:21:36 Retryable error: rpc error: code = DeadlineExceeded desc = Synthetic error: partial failure of ApplyBulk, retrying in 33.410051ms
2017/04/22 21:21:36 Retryable error: rpc error: code = DeadlineExceeded desc = Synthetic error: partial failure of ApplyBulk, retrying in 194.953937ms
2017/04/22 21:21:37 Retryable error: rpc error: code = DeadlineExceeded desc = Synthetic error: partial failure of ApplyBulk, retrying in 128.55932ms
...
```

Looking in `ApplyBulk` I see row errors are indeed returned but retired indefinitely. Debugging showed the actual error as `Internal` with the description `unknown family "family"`. However, this error is never returned and doesn't even appear in the logs. 

I expect a meaningful error to come back if the column family does not exist and is not automatically created. For example if a table does not exist you get back a `NotFound` error. Moreover the idempotent retry codes are defined as: 

```
idempotentRetryCodes  = []codes.Code{codes.DeadlineExceeded, codes.Unavailable, codes.Aborted, codes.Internal}
``` 
which includes `Internal`. 

Is `Internal` the correct error when a column family does not exist? Is it correct to retry an `Internal` error? 

For completeness I am also including the `makeClient` function:

```
func makeClient(t *testing.T) (*bigtable.AdminClient, *bigtable.Client, func()) {
	srv, err := bttest.NewServer("127.0.0.1:0")
	if err != nil {
		t.Fatalf("creating bttest.Server: %v", err)
	}

	ctx := context.Background()

	conn, err := grpc.Dial(srv.Addr, grpc.WithInsecure())
	if err != nil {
		t.Fatalf("dialing bttest.Server: %v", err)
	}

	proj, instance := "proj", "instance"

	adminClient, err := bigtable.NewAdminClient(ctx, proj, instance, option.WithGRPCConn(conn))
	if err != nil {
		t.Fatalf("creating admin client: %v", err)
	}
	client, err := bigtable.NewClient(ctx, proj, instance, option.WithGRPCConn(conn))
	if err != nil {
		t.Fatalf("creating client: %v", err)
	}
	return adminClient, client, func() {
		adminClient.Close()
		client.Close()
		conn.Close()
	}
}
```

**Comment by garye** [comment]
Thanks for detailed bug report. A few things:

- It's not good to retry on `Internal` errors, but we're doing so right now to work around a known issue in gRPC where a retryable error is mis-mapped to `Internal`. Once the gRPC bug is fixed we'll fix this library.

- Missing column families should result in a `NotFound` error not `Internal`, but as far as I can see this is only a bug in the emulator / inmem test server that the unit tests use. Have you seen this in production? I couldn't reproduce against an actual instance with your test case.

- The emulator bug(s) will be fixed.

**Comment by Iulian7** [comment]
Thanks for the info, it all makes sense. I have not seen seen this in production - sorry I should have mentioned it happens with the test server only. Still I am looking forward to seeing the bugs in gRPC and the emulator fixed.

**Comment by garye** [comment]
Note, the `Internal` retry fix is blocked by https://github.com/grpc/grpc-go/pull/1087

**Comment by tomwilkie** [comment]
grpc/grpc-go#1087 is closed now - can we remove Internal from the list of idempotentRetryCodes?

**Comment by garye** [comment]
Seems that way. I'll get a CL ready.

**Comment by tomwilkie** [comment]
Thanks!

**Comment by garye** [comment]
Fixed via e432073
